tools/release.sh

```diff
@@ -42,27 +44,27 @@ shift $((OPTIND-1))
 echo "# Selecting GPG key ..."
 
 gpgkey=$(gpg --list-secret-keys --keyid-format SHORT | awk -F'( +|/)' '/^(sec|ssb)/{print $3}')
-keycount=$(echo $gpgkey | wc -w)
+keycount=$(echo "$gpgkey" | wc -w)
 
-if [ $keycount -eq 0 ]; then
-  echo 'Need at least one GPG key, please make one with `gpg --gen-key`'
-  echo 'You will also need to submit your key to a public keyserver, e.g.'
-  echo '  https://sks-keyservers.net/i/#submit'
+if [ "$keycount" -eq 0 ]; then
+  echo "Need at least one GPG key, please make one with `gpg --gen-key`"
+  echo "You will also need to submit your key to a public keyserver, e.g."
+  echo "  https://sks-keyservers.net/i/#submit"
   exit 1
-elif [ $keycount -ne 1 ]; then
-  echo -e 'You have multiple GPG keys:\n'
+elif [ "$keycount" -ne 1 ]; then
+  printf "You have multiple GPG keys:\n"
 
   gpg --list-secret-keys
 
   while true; do
-    echo $gpgkey | awk '{ for(i = 1; i <= NF; i++) { print i ") " $i; } }'
-    echo -n 'Select a key: '
-    read keynum
-
-    if $(test "$keynum" -eq "$keynum" > /dev/null 2>&1); then
-      _gpgkey=$(echo $gpgkey | awk '{ print $'${keynum}'}')
-      keycount=$(echo $_gpgkey | wc -w)
-      if [ $keycount -eq 1 ]; then
+    echo "$gpgkey" | awk '{ for(i = 1; i <= NF; i++) { print i ") " $i; } }'
+    printf 'Select a key: '
+    read -r keynum
+
+    if [ "$keynum" -eq "$keynum" ]; then
```

Comment:
Wow, an actual `test` command 😮  I thought actual `test`s instead of `[]`s are mythical creatures from the 80s.

Reply:
In all likelihood all our tests should be enhanced tests `[[]]`s instead by the way. `test` and the `[` alias is pretty confusing and has weird edge cases.

Reply:
`[[]]` is a bashism:

```
In …/node/tools/release.sh line 65:
    if [[ "$keynum" -eq "$keynum" ]]; then
       ^---------------------------^ SC2039: In POSIX sh, [[ ]] is undefined.
```